sdk/cosmos/azure-cosmos/azure/cosmos/_retry_utility.py

```diff
@@ -112,6 +129,33 @@ def Execute(client, global_endpoint_manager, function, *args, **kwargs):
                 retry_policy = sessionRetry_policy
             elif exceptions._partition_range_is_gone(e):
                 retry_policy = partition_key_range_gone_retry_policy
+            elif exceptions._container_recreate_exception(e):
+                retry_policy = container_recreate_retry_policy
+                # pylint: disable=protected-access
+                # Before we retry if retry policy is container recreate, we need refresh the cache of the
+                # container properties and pass in the new RID in the headers.
+                client._refresh_container_properties_cache(retry_policy.container_link)
+                if retry_policy.check_if_rid_different(
+                        retry_policy.container_link, client._container_properties_cache, retry_policy.container_rid):
+                    retry_policy.refresh_container_properties_cache = False
+                else:
+                    cached_container = client._container_properties_cache[retry_policy.container_link]
+                    # If partition key value was previously extracted from the document definition
+                    # reattempt to extract partition key with updated partition key definition
+                    if retry_policy.should_extract_partition_key(args[3].headers, cached_container):
+                        new_partition_key = retry_policy._extract_partition_key(
+                            client, container_cache=cached_container, body=args[3].body
+                        )
+                        args[3].headers[HttpHeaders.PartitionKey] = new_partition_key
+                    # If getting throughput, we have to replace the container link received from stale cache
+                    # with refreshed cache
+                    if retry_policy.should_update_throughput_link(args[3].body, cached_container):
+                        new_body = retry_policy._update_throughput_link(args[3].body)
+                        args[3].body = new_body
+                        args[3].data = new_body
```

Comment:
hmmm this looks like fragile code to me - I think there's already a workitem to update the HttpRequest object used in this SDK, and setting the fields like this may be problematic. Also not sure why you're setting both of them.

Reply:
In this case body and data contain the same information for the request (its a string of the container link). Unfortunately, the code is fragile so it might require some updates if any changes to httprequest affects it. This is an edge case for container recreate.